numba/tests/test_npdatetime.py

```diff
@@ -541,7 +541,12 @@ def check(a):
     def test_hash(self):
         f = self.jit(hash_usecase)
         def check(a):
-            self.assertPreciseEqual(f(a), hash(a))
+            # Generic timedeltas (without usage of units)
+            # is no longer hashable beyond NumPy 2.2
+            # Non-generic timedeltas will have dtype name
+            # as timedelta64[<unit>]
+            if numpy_version < (2, 2) and a.dtype.name != 'timedelta':
```

Comment:
```suggestion
            if numpy_version < (2, 2) and a.dtype.name != 'timedelta64':
```
or ideally, match on the `dtype` and not the name?